### DISCUSSION TITLE
Include ibm.spectrum_virtualize collection

### DISCUSSION THREAD

**Discussion by Shilpi-J** [ansible-collections/ansible-inclusion#35] [Resolved reviews]
Intent is to include ibm.spectrum_virtualize collections in Ansible community release. Please validate and provide your feedback for the same.
Ansible Galaxy: https://galaxy.ansible.com/ibm/spectrum_virtualize
GitHub: https://github.com/ansible-collections/ibm.spectrum_virtualize
Issues Tracker: https://github.com/ansible-collections/ibm.spectrum_virtualize/issues
GitHub Handles: 
Is the collection part of Automation Hub: Yes
We meet [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst): Yes

**Comment by gundalow** [comment]
https://github.com/ansible-collections/ibm.spectrum_virtualize/blob/develop/meta/runtime.yml#L2 says the collection supports Ansible 2.9 (and higher), though CI testing doesn't include 2.9 https://github.com/ansible-collections/ibm.spectrum_virtualize/blob/develop/.github/workflows/main.yml#L15-L18

**Reply by Shilpi-J** [reply]
Thanks John, we are working on this.

**Reply by Shilpi-J** [reply]
We have included sanity checks against 2.9 as well in CI

**Comment by Shilpi-J** [comment]
Please validate and share your feedback.

**Comment by Andersson007** [comment]
# Ansible Collections Checklist (short version)

_For details about the following points, refer to the [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst)._

Every comment should say whether the reviewer expects it to be addressed, or whether it's optional.

**Public availability and communication:**
- [x] published on [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
  **RECOMMENDED**: put the link in README
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public SCM repository, releases of the collection are tagged in this repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
  **MUST FIX**: https://github.com/ansible-collections/ibm.spectrum_virtualize/pull/52
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules that only gather information are named `<something>_info`
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [x] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  - [x] `check_mode` is supported in all `_info` and `_facts` modules
- [x] supports Python 2.6 or greater and Python 3.5 or greater. If it does not, read the [full guidelines](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
- [x] only uses the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] documentation, examples, and return sections use FQCNs for the `M(..)` [format macros](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) when referring to modules
  **COMMENT:** See https://github.com/ansible-collections/ansible-inclusion/discussions/35#discussioncomment-2171713
- [x] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [x] FQCNs are used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-core

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`

**Tests:**
- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it does not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests run against every pull request
- [x] all CI tests run against a commit that releases the collection; if they don't pass, the collection won't be released
- [x] all CI tests run regularly (nightly, or at least once per week)
- [ ] according to the [requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#requirements-for-collections-to-be-included-in-the-ansible-package), collections to be included in the package must have unit or integration tests
  **MUST FIX**: Add unit or integration tests

**Reply by Andersson007** [reply]
@Shilpi-J i've finished the first round of review. See the above and https://github.com/ansible-collections/ibm.spectrum_virtualize/pull/52.
For me, the major issue is the last must fix, i.e. absence of unit or integration tests.

**Reply by Shilpi-J** [reply]
@Andersson007 The collection does have unit tests.https://github.com/ansible-collections/ibm.spectrum_virtualize/tree/develop/tests/unit/plugins/modules
It would be great if you can let me know what else is expected here?

**Reply by gundalow** [reply]
Appears that Unit tests aren't currently been run, please copy https://github.com/ansible-collections/collection_template/blob/main/.github/workflows/ansible-test.yml#L81-L134 into your GitHub Action.

**Reply by Andersson007** [reply]
@Shilpi-J I wasn't clear, sorry, as John spotted they should be included in the GHA workflow

**Reply by Shilpi-J** [reply]
Thanks @Andersson007 
I got the point, will add them in the workflow.

**Comment by Andersson007** [comment]
@Shilpi-J what do you think of simplifying the examples, i.e. instead of
```
 92 - name: Using Spectrum Virtualize collection to create an empty host cluster
 93   hosts: localhost
 94   collections:
 95     - ibm.spectrum_virtualize
 96   gather_facts: no
 97   connection: local
 98   tasks:
 99     - name: Define a new host cluster
100       ibm_svc_hostcluster:
101         clustername: "{{clustername}}"
102         domain: "{{domain}}"
103         username: "{{username}}"
104         password: "{{password}}"
105         log_path: /tmp/playbook.debug
106         name: hostcluster0
107         state: present
108         ownershipgroup: group1
```
something like
```
 99 - name: Define a new host cluster
100   ibm.spectrum_virtualize.ibm_svc_hostcluster:
101     clustername: "{{clustername}}"
102     domain: "{{domain}}"
103     username: "{{username}}"
104     password: "{{password}}"
105     log_path: /tmp/playbook.debug
106     name: hostcluster0
107     state: present
108     ownershipgroup: group1
```

**Reply by Shilpi-J** [reply]
@Andersson007 Yes we can do that..

**Comment by Shilpi-J** [comment]
@Andersson007 We have incorporated your suggestions in our collection. The new release is out with all these changes.
Please validate and share your feedback.
Thanks

**Reply by Andersson007** [reply]
@Shilpi-J thanks! I'll try to take a look as soon as possible.

**Comment by Andersson007** [comment]
@Shilpi-J the collection LGTM now, thanks for fixing things, I'll create a corresponding community topic and will ask steering committee folks to conduct the second necessary review. Thanks!

**Reply by Andersson007** [reply]
https://github.com/ansible-community/community-topics/issues/85

**Reply by Shilpi-J** [reply]
Thanks @Andersson007

**Comment by cybette** [comment]
Review by @cybette [done]

# Ansible Collections Checklist (short version)

_For details about the following points, refer to the [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst)._

Every comment should say whether the reviewer expects it to be addressed, or whether it's optional.

**Public availability and communication:**
- [x] published on [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public SCM repository, releases of the collection are tagged in this repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules that only gather information are named `<something>_info`
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [x] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  - [x] `check_mode` is supported in all `_info` and `_facts` modules
- [x] supports Python 2.6 or greater and Python 3.5 or greater. If it does not, read the [full guidelines](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
- [x] only uses the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] documentation, examples, and return sections use FQCNs for the `M(..)` [format macros](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) when referring to modules
- [x] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [x] FQCNs are used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-core
- [x] public plugins, roles and playbooks do not use files outside of `meta/`, `plugins/`, `roles/`, and `playbooks/`

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`
- [x] collection repository should not contain any large objects (binaries) comparatively to the current Galaxy tarball size limit of 20 MB. For example, package installers for testing purposes shouldn't be added.
- [x] collection repository should not contain any unnecessary files like, for example, temporary files created by development tools. Temporary files MUST be added to `.gitignore`.

**Tests:**

- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it does not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests run against every pull request
- [x] all CI tests run against a commit that releases the collection; if they don't pass, the collection won't be released
- [x] all CI tests run regularly (nightly, or at least once per week)

**Reply by Shilpi-J** [reply]
@cybette @Andersson007 Is the review complete? Do you see anything which is pending from our side? Our collection new release 1.9.0 is out. Let me know if our collection is ready for the inclusion. 
Thanks

**Reply by cybette** [reply]
Hi @Shilpi-J , apologies for the delay as I have been held up by PTO and some personal obligations. I will have the review completed within the next week. Thanks for your patience and understanding!

**Reply by Shilpi-J** [reply]
Thanks @cybette , waiting for your feedback.

**Comment by cybette** [comment]
LGTM! Once again sorry for taking so long and thanks for the great cooperation and contribution!

**Reply by Shilpi-J** [reply]
Thanks Carol.
@Andersson007 @cybette Since both the reviews are complete, what would be the next step? How soon can we expect our collection to be included?

**Reply by Andersson007** [reply]
@Shilpi-J I would expect that it'll happen within ~2 weeks since now

**Reply by Shilpi-J** [reply]
Thanks @Andersson007

**Comment by gundalow** [comment]
Approved, thanks all

**Reply by Shilpi-J** [reply]
Thanks @gundalow

**Comment by Andersson007** [comment]
@Shilpi-J https://github.com/ansible-community/ansible-build-data/pull/148 has been merged.
The collection will appear in the package in the next 6.X minor release (which happens every 3 weeks).
Thanks everyone!

**Reply by Shilpi-J** [reply]
This is great news ! Thanks @Andersson007
